Correction of the behavior to check a module name

nusphere opened the pull request:
|    Q          |   A
|-------------- | ------
| Bugfix        | yes (#20)

### Description

In this merge request, the current behavior is represented by a test case.

A module that lies within a namespace with more than 1 level is not correctly checked by the "assertModuleName".


fixes #20

nusphere committed:
Add Testcases for loading modules with namespaces

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere committed:
Add Testcases for loading modules with namespaces

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere committed:
Bugfix - Find correct module name

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere committed:
introduce a clean assertation with clean contraint - described at #6

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere commented:
I have revised the entire assert function again so that it is properly provided as a constraint. The test coverage was used to check whether all previous behavior was retained.

Using Constrains enables the actual test to be reused cleanly.

nusphere committed:
using parent::fail to keep backward compatibility promise for PHPUnit

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere committed:
solving code style issues

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

froschdesign commented on src/PHPUnit/Controller/AbstractControllerTestCase.php:
```diff
@@ -536,40 +539,21 @@ protected function getControllerFullClassName()
-     * @param string $module
      */
-    public function assertModuleName($module)
+    final public function assertModuleName(string $module): void
```
If you change the public API then this is a BC break. Please revert this, otherwise the changes will have to wait until the next main version.

froschdesign commented on src/PHPUnit/Controller/AbstractControllerTestCase.php:
```diff
@@ -536,40 +539,21 @@ protected function getControllerFullClassName()
-     * @param string $module
      */
-    public function assertNotModuleName($module)
+    final public function assertNotModuleName(string $module): void
```
The same here.

nusphere replied:
Resolved

nusphere committed:
Revert BC breaks

Signed-off-by: Sebastian Hopfe <nusphere83@gmail.com>

nusphere commented:
both BC breaks are solved.

weierophinney committed:
qa: fix as many Psalm errors as possible

Most are due to usage of `AbstractControllerTestCase::createFailureMessage()`, which is marked deprecated, but which we cannot move our own code away from due to BC implications.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

nusphere commented:
thank you
